Searchtool

arowla opened the pull request:
I cleaned up a ton of unused dependencies from the vendor directory and some unused static assets-- at least the things that I could easily confirm weren't used. There are probably still lots of JS and CSS files included in the layout that aren't actually called upon, judging by how much is there.

I extracted a couple of elements into a new file, `server/server-config.js`, and used them to simplify the `server/boot/routes.js` file. The login hack is no longer necessary, and the Solr URL is no longer hardcoded in multiple places.

This is all in addition to doing a basic jshint cleanup on the routes.js file.

cc @noonkay -- would be nice to get a review-- don't worry about going over everything with a fine toothed comb, but at least if you can look over my last couple commits with refactors, that would be great.

darrenculbreath committed:
Init Push of Search code

Commit:
Fixed alignment and jshint warnings

Commit:
Removed unused vendored libraries

Commit:
Removing unused images

Commit:
Removing unused icons

Commit:
Add missing dependency, unirest

Commit:
Extract server config to file

- Allow required logins to be turned off for development
- Put Solr URI in config -- don't hardcode

arowla commented:
LOL Hound.

neka-io commented:
:tada: This does look much better! Two things that come to mind:
1. We could do a `bower init` to create a `bower.json` that can manage and install these dependencies in a cleaner way. The bower components would be git ignored, just as the node modules are. As an added bonus, we wouldn't hear from Hound about third party code style violations! :smile: 
2. I'd consider moving buildSearch function to a different file for readability and separation of concerns.

Commit:
Move dependencies from client/vendor to bower

arowla commented:
I moved the dependencies to bower. Now there are only 45 files changed. :grinning:

Commit:
Fix login logic

- If login is required, check for the cookie, then parse it. Check the
  cookie ID again. If either of these steps fails, redirect to login.
- Finally, build the search.

Commit:
Refactored buildSearch into separate file
